AutomationServices/AutoMailChecker/utils.js

```diff
@@ -1,34 +1,67 @@
 const mailparser = require('mailparser').simpleParser
 const _ = require('lodash')
 const cheerio = require('cheerio');
+const { htmlToText } = require('html-to-text');
 
 
-module.exports.readRawEmail = async (emails=[]) =>{
+function isBase64(str) {
+    return Buffer.from(str, 'base64').toString('base64') === str
+}
+
+module.exports.readRawEmail = async (emails = []) => {
     const parsed = []
     for (const mail of emails) {
-        const all = _.find(mail.parts, { "which": "TEXT" })
-        const result = await mailparser(all.body);
-        const subject = mail.parts.filter(part=>part.which==='HEADER')[0].body.subject[0]
+        let all = _.find(mail.parts, { "which": "TEXT" })
+        const subject = mail.parts.filter(part => part.which === 'HEADER')[0].body.subject[0]
+        let textAsHtml = null;
+
+        if (all.body !== null && isBase64(all.body.replace(/\r?\n|\r/g, ""))) {
+            all.body = Buffer.from(all.body.replace(/\r?\n|\r/g, ""), 'base64').toString()
+            textAsHtml = '<p>' + htmlToText(all.body).replace(/\r?\n|\r|\t/g, " ") + '</p>'
+        } else {
+            const result = await mailparser(all.body);
+            textAsHtml = result.textAsHtml
+        }
+
 
         parsed.push({
             date: mail.attributes.date,
             subject: subject,
-            html: result.textAsHtml
+            html: textAsHtml
         })
+
     }
     return parsed
 }
 
-module.exports.search = (html, filter, skipped_phrase = 'Bancolombia le informa que su factura inscrita') =>{
+module.exports.search = (html, filter, skipped_phrase = 'Bancolombia le informa que su factura inscrita') => {
     const $ = cheerio.load(html)
     const res = $('p')
     const value = res.text().trim()
-    if(value.indexOf(filter) > 0){
-        if(value.indexOf(skipped_phrase) < 0){
+    if (value.indexOf(filter) > 0) {
+        if (value.indexOf(skipped_phrase) < 0) {
+            let description, textWithValue;
             const first = value.indexOf(filter)
-            const end = value.indexOf('*', first + 1)
-            const newValue = value.substring(first, end)
-            return newValue
+
+            if (value.indexOf('*', first + 1) > 0) {
```

Comment:
~are you detecting the bancolombia bank message using the **56 for the account?? what happen in the case where appears the market name? 🤔🤔🤔~

Nevermind, the previous comment was before seeing that i wrote that piece of code :) 

What do you think about detect through the bank.name passed by params?